### PR TITLE
Updated development-related documentation

### PR DESCRIPTION
**Pull request by evildmp**

### CODE REVIEW

File: docs/contributing/index.rst [hunk 1]
```diff
@@ -89,15 +89,35 @@ versions are maintained through the community. Divio provides long term support
...
-* **develop**: this will be *django CMS 3.3*; this is the most appropriate
-  branch for more substantial features that will need team co-ordination
+We maintain a number of branches on our GitHub repository.
+
+
+
+the latest (highest-numbered) ``release/x.y.z``
+    This is the branch that will become the next release on PyPI.
+
+    **Fixes and backwards-compatible improvements** (i.e. the vast majority of pull requests) should be made against
```
Comment (timgraham) [inline docs/contributing/index.rst]: You could clarify how these changes make it to the "develop" branch (i.e. the committer cherry-picks or merges it)? This is different from Django's workflow where all PRs go to master and only those issues that qualify for a backport to the stable release are cherry-picked by the committer.

**Comment by evildmp** [comment]
I would expect that the latest *release* branch would be merged into *develop*. @mkoistinen comments?

(And did I really put in three carriage returns? Weird.)

**Comment by mkoistinen** [comment]
@evildmp Typically what *does* happen, is that when a new release is being prepared, someone simply merges forward everything from the "source" branch. E.g., release/3.2.x is merged to develop before releasing 3.3.0.

What *should* happen, is that developers are cognizant of the features they are writing, and **start** on the correct branch for the nature of the change, then forward- and back-port as required. It is daunting to be preparing a major release, and to have a pile of commits you have no experience with to tease-apart when there are merge-conflicts.

This is why I 👍 'ed @timgraham's comment. Not only is clarity required here in these docs, but also we should probably record, follow and enforce a policy that the person doing the original commit is also responsible for forward- and down-porting as required **at the time of the original commit**. (Note, I'll probably be the first to violate this, since my fingers just 'uttered' it. =/)

Exceptions, of course, are made when starting a new develop branch, and also for excepting "hit and run" commits from the community, etc., but I would think that the core developers wouldn't be so easily exempt =).

**Comment by yakky** [comment]
I personally am a fan of mass forwardporting from earlier releases. While resolving merge conflicts it's not  fun, it's a much less error prone process than cherry-picking commits. In Django project is easier as each PR it always consist of a single commit, in django CMS this does not happen and porting entire branches can be quite painful and time consuming process, and the risk of losing commits in the process is quite high

**Comment by timgraham** [comment]
Are you opposed to squashing commits when merging commits? In my experience, browsing django-cms with something like `git log` is rather useless since there are many commits like "remove emty line" or "fix linting". Just wondering if you find value in continuing this workflow. If it makes any difference and as you may know, GitHub has added the "squash and merge" button.

**Comment by yakky** [comment]
@timgraham not opposing at all. I'm just reporting the current situation and the effect of porting branches around

**Comment by mkoistinen** [comment]
I'm not fundamentally opposed to retaining the mass forward-, back-porting and I do think it would be easier/clearer if we used squashed commits more. I have indeed been enjoying the new GitHub feature.

**Comment by evildmp** [comment]
It seems fundamentally easier and more friction-free, at least in principle, to merge things forwards rather than backwards. 

It's also easier in practice for django CMS, since we do tend to have a messy commit history, and as @timgraham says it's not just aesthetically messy, it actually makes working with the code base harder than it ought to be.

Are there sound reasons for preferring merging forward or cherry-picking backwards? Why does Django do it the way it does?

**Comment by timgraham** [comment]
I think Django is more conservative about making changes in stable releases (only backporting certain issues as described in our [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions)). django-cms seems to also accept small features and enhancements in the latest stable release branch.

**Comment by yakky** [comment]
actually most of the commits in stable branches are bugfix, but in django CMS "bug" has a much broader sense than in django, mostly because it's an end-user facing project, and even small bugs must be fixed in stable releases and cannot wait a major release.
Being a smaller project with a smaller team, also, process must be more nimbler to allow bending rules, if needed

**Comment by timgraham** [comment]
The proposed change here to say "the vast majority of pull requests (are against the release branch)" might not be accurate.

**Comment by vxsx** [comment]
👍

**Comment by FinalAngel** [comment]
👍 💯

**Comment by evildmp** [comment]
Thanks all for the feedback, I've made a number of changes and additions in light of them.

**Comment by czpython** [comment]
👍  :shipit:

File: docs/contributing/index.rst [hunk 2]
```diff
@@ -89,19 +94,132 @@ versions are maintained through the community. Divio provides long term support
...
 
 
-*************
+.. _branch_policy:
+
 Branch policy
-*************
+=============
+
+.. versionchanged:: 3.3
```
Comment (timgraham) [inline docs/contributing/index.rst]: As a reader, I don't think these annotations provide much value unless you describe what changed. (ala https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#documenting-new-features)

File: docs/contributing/index.rst [hunk 3]
```diff
@@ -89,19 +94,132 @@ versions are maintained through the community. Divio provides long term support
...
+    **Fixes and backwards-compatible improvements** (i.e. most pull requests) will be made against
+    this branch.
 
-If in doubt, ask on the #django-cms IRC channel on `freenode`_ or the
+``develop``
+    This is the branch that will become the next release that increments the ``x`` or ``y`` of the latest
+    ``release/x.y.z``.
+
+    This branch is for **new features and backwards-incompatible changes**. By their nature, these will require more
+    substantial team co-ordination.
```
Comment (timgraham) [inline docs/contributing/index.rst]: I've never seen coordination spelled with a dash. Is it intentional?

**Comment by mkoistinen** [comment]
LGTM, merging

File: docs/contributing/index.rst (hunk 3 above)
Comment (evildmp) [inline docs/contributing/index.rst]: Standard in UK English - co-ordination, co-operation, etc. The cool kids use cördination and cöperation, but I wouldn't have the nerve.

File: docs/contributing/index.rst (hunk 2 above)
Comment (evildmp) [inline docs/contributing/index.rst]: Thanks, have taken a note.

**Pull request merged**